src/main/java/edu/kis/powp/jobs2d/visitor/MirrorFigureCommandVisitor.java

```diff
@@ -0,0 +1,46 @@
+package edu.kis.powp.jobs2d.visitor;
+
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+import java.util.Iterator;
+
+public class MirrorFigureCommandVisitor implements ICommandVisitor{
+    private final boolean mirrorHorizontal;
+    private final boolean mirrorVertical;
+
+    public MirrorFigureCommandVisitor(boolean mirrorHorizontal, boolean mirrorVertical){
+        this.mirrorHorizontal = mirrorHorizontal;
+        this.mirrorVertical = mirrorVertical;
+    }
+
+    @Override
+    public void visit(OperateToCommand command) {
+        if (mirrorHorizontal){
+            command.setPosX(command.getPosX() * -1);
+        }
+        if(mirrorVertical){
+            command.setPosY(command.getPosY() * -1);
+        }
```

Comment:
I would prefer the immutable approach, i,e., creating new commands from originals (no setters).

Reply:
During development, we decided to stay with setters, because other implementations would require heavy changes in code that would result in less readable code and potential conflicts.

Reply:
of course, it depends on the changes you are planning. Still you should not think that you will avoid "heavy changes" ;)